nixos/modules/services/databases/cockroachdb.nix

```diff
@@ -0,0 +1,170 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.cockroachdb;
+
+  cockroachdb = cfg.package;
+
+  escape = builtins.replaceStrings ["%"] ["%%"];
+
+  secureArg = insecure: certsDir:
+    if insecure then "--insecure " else
+    if (!isNull certsDir) then "--certs-dir=${certsDir} " else
+    throw  "error: cockroachdb must specify either insecure or certsDir";
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.cockroachdb = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "
+          Whether to enable the CockroachDB server.
+        ";
+      };
+
+      package = mkOption {
```

Comment:
It seems pretty standard to include this option. I think it's useful enough to be able to override the package with a user-supplied package (maybe you need 2.1 plus commit `xyz`). Yes, you could also override the service with a user-supplied service. I can't say whether or not it's worth the extra option, but the harm in having it seems minimal since it doesn't have any significant code to go with it.

Reply:
On second thought, Cockroach has one important distinction that separates it from the other stuff: it has the fully FOSS version and the "Cockroach Community License" (CCL) version with Enterprise features. It's reasonable to assume someone might want, and build, an enterprise copy for themselves. (Whether or not they'd use this module to configure it is a different matter, though).